src/main/java/org/openjdk/jextract/impl/StructBuilder.java

```diff
@@ -282,70 +292,78 @@ private void emitFieldArraySetter(String javaName, Declaration.Variable varTree,
         appendBlankLine();
         emitFieldDocComment(varTree, "Indexed setter for field:");
         if (Utils.isStructOrUnion(elemType)) {
-            appendIndentedLines(STR."""
-                public static void \{javaName}(MemorySegment \{segmentParam}, \{indexList.decl()}, MemorySegment \{valueParam}) {
-                    MemorySegment.copy(\{valueParam}, 0L, \{javaName}(\{segmentParam}, \{indexList.use()}), 0L, \{layoutString(elemType)}.byteSize());
+            appendIndentedLines(String.format("""
+                public static void %s(MemorySegment %s, %s, MemorySegment %s) {
+                    MemorySegment.copy(%s, 0L, %s(%s, %s), 0L, %s.byteSize());
                 }
-                """);
+                """,
+                javaName, segmentParam, indexList.decl(), valueParam, valueParam,
+                javaName, segmentParam, indexList.use(), layoutString(elemType)));
         } else {
-            appendIndentedLines(STR."""
-                public static void \{javaName}(MemorySegment \{segmentParam}, \{indexList.decl()}, \{elemTypeCls.getSimpleName()} \{valueParam}) {
-                    \{arrayElementHandle}.set(\{segmentParam}, 0L, \{indexList.use()}, \{valueParam});
+            appendIndentedLines(String.format("""
+                public static void %s(MemorySegment %s, %s, %s %s) {
+                    %s.set(%s, 0L, %s, %s);
                 }
-                """);
+                """,
+                javaName, segmentParam, indexList.decl(), elemTypeCls.getSimpleName(),
+                valueParam, arrayElementHandle, segmentParam, indexList.use(), valueParam));
         }
     }
 
     private void emitAsSlice() {
         String arrayParam = safeParameterName("array");
-        appendIndentedLines(STR."""
+        appendIndentedLines(String.format("""
 
             /**
-             * Obtains a slice of {@code arrayParam} which selects the array element at {@code index}.
-             * The returned segment has address {@code arrayParam.address() + index * layout().byteSize()}
+             * Obtains a slice of %s which selects the array element at %s.
+             * The returned segment has address %s.address() + index * layout().byteSize()
              */
-            public static MemorySegment asSlice(MemorySegment \{arrayParam}, long index) {
-                return \{arrayParam}.asSlice(layout().byteSize() * index);
+            public static MemorySegment asSlice(MemorySegment %s, long index) {
+                return %s.asSlice(layout().byteSize() * index);
             }
-            """);
+            """,
+            arrayParam, arrayParam, arrayParam, arrayParam, arrayParam));
```

Comment:
In this case, using a `MessageFormat`-like approach is particularly convenient (as there would be just one param)